Use pkg.installed not latest

noelmcloughlin opened the pull request:
Resolves #111

aboe76 commented:
@noelmcloughlin in #111 there is also the point of version or update

in https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pkg.html#salt.states.pkg.installed
the option to add version and allow_updates: True

noelmcloughlin commented:
@aboe76 I never noticed, thanks I have reviewed and updated the commit.  

I have introduced new parameters under new `docker:pkg` namespace - but left depreciated `docker-pkg:version` pillar intact - future PR needs to cleanup this state.

aboe76 commented on docker/init.sls:
```diff
@@ -85,8 +85,8 @@ docker package repository:
+  {%- if "version" in docker %}
     {%- if grains["oscodename"]|lower == 'jessie' %}
     - name: docker.io
     - version: {{ docker.version }}
```
isn't this the same as  docker.pkg.version ?

noelmcloughlin replied:
No, that `docker` is the construct created by map.jinja which calculates `version` from the `docker-pkg:version` pillar. But `docker-pkg` namespace is, or should be, depreciated.

There is some repetition in the code block - I could fix that but small PR gets too big then?

noelmcloughlin commented:
Fixed conflicts.
